File: API-ShareDocuments/Domain/Enums/DocumentStatus.cs
using System.Text.Json.Serialization;

namespace Domain.Enums
{
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum DocumentStatus
    {
        Pending = 1,
        Published = 2, 
        Rejected = 3,
        Hidden = 4
    }
}

File: UI-ShareDocuments/src/common/constants/document_status.ts
export const DocumentStatus = {
    Pending: "Pending",
    Published: "Published",
    Rejected: "Rejected",
} as const;

export type DocumentStatus = (typeof DocumentStatus)[keyof typeof DocumentStatus];

export const DOCUMENT_STATUS_LABEL: Record<DocumentStatus, string> = {
    Pending: "Đang chờ",
    Published: "Đã xuất bản",
    Rejected: "Bị từ chối",
};

export const DOCUMENT_STATUSES: DocumentStatus[] = [
    DocumentStatus.Pending,
    DocumentStatus.Published,
    DocumentStatus.Rejected,
];

